format_func/src/lib.rs

```diff
@@ -34,28 +45,28 @@ pub fn format_5(
     value3: &str,
     value4: &str,
     value5: &str,
-) -> String {
+) -> Result<String, String> {
     let values = &[value1, value2, value3, value4, value5];
     return format(template, values);
 }
 
-pub fn format(template: &str, values: &[&str]) -> String {
+pub fn format(template: &str, values: &[&str]) -> Result<String, String> {
     let mut vars: HashMap<String, &str> = HashMap::default();
     for (i, value) in values.iter().enumerate() {
         vars.insert((i + 1).to_string(), value);
     }
     match strfmt(template, &vars) {
-        Ok(result) => result,
+        Ok(result) => Ok(result),
         Err(e) => {
             let error_msg_vals = values
                 .iter()
                 .map(|v| v.to_string())
                 .collect::<Vec<_>>()
                 .join(", ");
-            format!(
+            Err(format!(
                 "Formatting with template {} and values {} caused an error: {:?}",
```

Comment:
May be better to quote the `template` and `values` to make it more easy to read, like
"Formatting with template '{}' and values '{}' caused an error: {:?}"

Reply:
Values as error message seems do not make sense. I removed it and the error message will focus on template and the error.